packages/viz/src/lib/Layer.ts

```diff
@@ -1,107 +1,107 @@
-  private _mvtLayerInstance: MVTLayer<string> | undefined;
-
-  constructor(source: string, styles = {}, options: LayerOptions = {}) {
-    const { mapOptions = {}, credentials = defaultCredentials } = options;
-
-    this._credentials = credentials;
-    this._mapsClientInstance = new Maps(this._credentials);
+  private _deckLayer: any;
 
-    this._layerSource = new Source(source);
-    this._layerStyles = new Style(styles);
-
-    this._layerOptions = Object.assign({}, defaultMapOptions, mapOptions);
-    this._layerInstantiation = this._mapsClientInstance.instantiateMapFrom(
-      buildInstantiationOptions({ mapOptions: this._layerOptions, mapSource: this._layerSource })
-    );
+  constructor(source: string | CARTOSource, styles = {}) {
+    this._source = this._buildSource(source);
+    this._styles = new Style(styles);
   }
 
-  public async setSource(source: string) {
-    const previousSource = this._layerSource;
+  /**
+   * Change a source to the current layer.
+   * A new map instantion and a replace of the layer will be fired
+   * @param source source to be set
+   */
+  public async setSource(source: string | CARTOSource) {
+    const previousSource = this._source;
 
-    this._layerSource = new Source(source);
-    this._layerInstantiation = this._mapsClientInstance.instantiateMapFrom(
-      buildInstantiationOptions({ mapOptions: this._layerOptions, mapSource: this._layerSource })
-    );
+    this._source = this._buildSource(source);
 
-    if (this._mvtLayerInstance) {
+    if (this._deckInstance) {
       await this._replaceLayer(previousSource);
     }
   }
 
+  /**
+   * Change the styles of the current layer.
+   * A new map instantion and a replace of the layer will be fired
+   * @param style style to be set
+   */
   public async setStyle(style: {}) {
-    const previousSource = this._layerSource;
+    const previousSource = this._source;
 
-    this._layerStyles = new Style(style);
+    this._styles = new Style(style);
 
-    if (this._mvtLayerInstance) {
+    if (this._deckLayer) {
       await this._replaceLayer(previousSource);
     }
   }
 
+  /**
+   * Add the current layer to a Deck instance
+   * @param deckInstance instance to add the layer to
+   */
   public async addTo(deckInstance: any) {
+    // get current layers
     const currentDeckLayers = deckInstance.props.layers;
-    const createdDeckLayer = await this.getDeckGLLayer();
+    // create the new layer
+    const layer = await this._createDeckGLLayer();
 
     deckInstance.setProps({
       layers: [
         ...currentDeckLayers,
-        createdDeckLayer
+        layer
       ]
     });
 
     this._deckInstance = deckInstance;
   }
 
-  public async getDeckGLLayer() {
-    const layerProperties = await this._getDeckGLLayerProperties();
-
-    this._mvtLayerInstance = new MVTLayer(layerProperties);
-    return this._mvtLayerInstance;
-  }
+  /**
+   * Method to create the Deck.gl layer
+   */
+  private async _createDeckGLLayer() {
```

Comment:
We have lost the public method `getDeckGLLayer`. I think we could keep a simple public getter on `this._deckLayer`, so a dev can a reference to the pure deck object on demand